src/components/Opinion/Opinion.tsx

```diff
@@ -1,49 +1,221 @@
+    } else {
+      setLikes((prev) => (prev > 0 ? prev - 1 : 0));
+      setLiked(false);
+      onUnlike(reviewId);
+    }
+  };
+
+  const toggleDislike = () => {
+    console.log(disliked);
+    if (disliked == null) return;
+    if (disliked === false) {
+      setDislikes((prev) => prev + 1);
+      setDisliked(true);
+      if (liked) {
+        setLikes((prev) => (prev > 0 ? prev - 1 : 0));
+        setLiked(false);
+      }
+      onDislike(reviewId);
+    } else {
+      setDislikes((prev) => (prev > 0 ? prev - 1 : 0));
+      setDisliked(false);
+      onUndislike(reviewId);
+    }
+  };
+
+  const handleReport = (content: string) => {
+    onReport(reviewId, content);
+  };
+
+  const handleEdit = async (opinionRating: number, opinionContent: string) => {
+    onEdit(opinionRating, opinionContent);
+  };
+
+  const handleDelete = async () => {
+    onDelete(reviewId);
+    onDeleteClose();
+  };
+
+  return (
+    <div className="opinion-container">
+      <div className="opinion">
+        <div className="header">
+          <div className="user-info">
+            <span className="username">{username}</span>
+            <span className="rating">· {rating / 2}/5 </span>
+            <FaStar className="star" />
+            {isCurrentUser && isCurrentUser === true && (
+              <div className="edit-delete-buttons">
+                <button
+                  type="button"
+                  onClick={onEditOpen}
+                  className="opinion-edit-button"
+                >
+                  <FaEdit />
+                </button>
+                <button
+                  type="button"
+                  onClick={onDeleteOpen}
+                  className="opinion-delete-button"
+                >
+                  <FaTrashAlt />
+                </button>
+              </div>
+            )}
+          </div>
+          <span className="date">
+            {new Date(createdAt).toLocaleDateString('sv-SE')}
+          </span>
+        </div>
+        <div className="content">
+          {content.split('\n').map((par) => (
+            <p key={Math.random()}>{par}</p>
+          ))}
+        </div>
+        <div className="footer">
+          <button
+            className="report-button"
+            type="button"
+            onClick={onReportOpen}
+          >
+            Report
+          </button>
+
+          <div className="icons">
+            <FaThumbsUp
+              className={`icon ${liked ? 'liked' : ''}`}
+              onClick={toggleLike}
+            />
+            <span className="likes-count">{likes}</span>
+            <FaThumbsDown
+              className={`icon ${disliked ? 'disliked' : ''}`}
+              onClick={toggleDislike}
+            />
+            <span className="dislikes-count">{dislikes}</span>
+          </div>
+        </div>
       </div>
+      {(isEditOpen || isReportOpen || isDeleteOpen) && (
+        <div className="modals">
+          <OpinionModal
```

Comment:
I think we should stay consisent with the way we are displaying the modals, so either move the add opinion modal here or move all of these modals outside - I think they should be in the same place. I would advise moving them all outside (maybe creating an OpinionList component that stores the opinion list), as when they are here, upon clicking the modal it "takes space" below an opinion, creating a weird effect when there is more than one opinion present 

Below is the weird effect shown (it moves the second opinion on the second screenshot)

![image](https://github.com/Java-DZGS/ingreedio/assets/65892459/8ace81e8-ce34-43e8-b4b0-ea85e7fa9040)
![image](https://github.com/Java-DZGS/ingreedio/assets/65892459/5ec312cb-6ad5-4808-8483-29710bed065a)